### PR TITLE
Feature/34576 helpdesk summary

### PR DESCRIPTION
**Pull request by GuyHarwood**
- implement school summary page 
- link on school home in guidance box, visible to helpdesk only
- guard route via role check
- enable login timestamp on TIO checks
- docs: add readme for check started function, as this was analysed as part of the work

### CODE REVIEW

File: admin/services/data-access/school-summary.data.service.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  const schoolIdParam = {
+    name: 'schoolId',
+    type: TYPES.Int,
+    value: schoolId
+  }
+  const sql = `
+    SELECT
+        MIN(convert(varchar, c.createdAt, 106)) as [Date],
+        COUNT(c.id) AS [PinsGenerated],
+        SUM(CASE cs.code WHEN 'COL' THEN 1 ELSE 0 END) as [LoggedIn],
```
Comment (jon-shipley) [inline admin/services/data-access/school-summary.data.service.js:45]: I'm not sure about this as it is counting the 'current' state.   I would count all complete checks in the LoggedIn figure.  It must be the case!  

I propose:

```sql
SELECT
    MIN(convert(varchar, c.createdAt, 106)) as [Date],
    COUNT(c.id) AS [PinsGenerated],
    SUM(CASE WHEN c.pupilLoginDate IS NOT NULL THEN 1 ELSE 0 END) as [LoggedIn],
    SUM(CAST(c.complete AS INT)) as [Complete]
    FROM
        [mtc_admin].[check] c
        INNER JOIN [mtc_admin].pupil p ON (p.currentCheckId = c.id)
        LEFT JOIN [mtc_admin].[checkStatus] cs ON (c.checkStatus_id = cs.id)
    WHERE p.school_id = @schoolId AND c.isLiveCheck = 1
    GROUP BY cast(c.createdAt as date);
```

File: admin/services/data-access/school-summary.data.service.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    FROM
+        [mtc_admin].[check] c
+        INNER JOIN [mtc_admin].pupil p ON (p.currentCheckId = c.id)
+        LEFT JOIN [mtc_admin].[checkStatus] cs ON (c.checkStatus_id = cs.id)
+    WHERE p.school_id = @schoolId AND c.isLiveCheck = 1
+    GROUP BY cast(c.createdAt as date)`
+  return sqlService.readonlyQuery(sql, [schoolIdParam])
+}
+
+service.getTioCheckData = async (schoolId) => {
```
Comment (jon-shipley) [inline admin/services/data-access/school-summary.data.service.js:56]: I would just spell this out: `getTryItOutCheckData`.  Not a blocker though :)

File: admin/views/helpdesk/school-summary.ejs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            <th scope="col" class="govuk-table__header govuk-!-width-one-quarter">
+              Date
+            </th>
+            <th scope="col" class="govuk-table__header govuk-!-width-one-quarter">
+              Pins Generated
+            </th>
+            <th scope="col" class="govuk-table__header govuk-!-width-one-quarter">
+              Users Logged In
+            </th>
+<!--             <th scope="col" class="govuk-table__header govuk-!-width-one-quarter">
```
Comment (jon-shipley) [inline admin/views/helpdesk/school-summary.ejs:54]: What's this?

File: admin/views/helpdesk/school-summary.ejs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+            <td scope="row">
+              <%= lcs.Date %>
+            </td>
+            <td class="govuk-table__cell">
+              <%= lcs.PinsGenerated %>
+            </td>
+            <td class="govuk-table__cell">
+              <%= lcs.LoggedIn %>
+            </td>
+<!--             <td class="govuk-table__cell">
```
Comment (jon-shipley) [inline admin/views/helpdesk/school-summary.ejs:71]: And this?

File: admin/views/helpdesk/school-summary.ejs [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            <th scope="col" class="govuk-table__header govuk-!-width-one-quarter">
+              Date
+            </th>
+            <th scope="col" class="govuk-table__header govuk-!-width-one-quarter">
+              Pins Generated
+            </th>
+            <th scope="col" class="govuk-table__header govuk-!-width-one-quarter">
+              Users Logged In
+            </th>
+<!--             <th scope="col" class="govuk-table__header govuk-!-width-one-quarter">
```
Comment (jon-shipley) [inline admin/views/helpdesk/school-summary.ejs:91]: What's this?

File: admin/views/helpdesk/school-summary.ejs [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            <td scope="row">
+              <%= tcs.Date %>
+            </td>
+            <td class="govuk-table__cell">
+              <%= tcs.PinsGenerated %>
+            </td>
+            <td class="govuk-table__cell">
+              <%= tcs.LoggedIn %>
+            </td>
+<!--             <td class="govuk-table__cell">
```
Comment (jon-shipley) [inline admin/views/helpdesk/school-summary.ejs:108]: And this?

File: admin/views/helpdesk/school-summary.ejs [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+      <div class="govuk-back-to-top">
+        <a class="govuk-link govuk-link--no-visited-state app-back-to-top__link" href="#top">
+          <svg role="presentation" focusable="false" class="app-back-to-top__icon" xmlns="http://www.w3.org/2000/svg"
+            width="13" height="17" viewBox="0 0 13 17">
+            <path fill="currentColor" d="M6.5 0L0 6.5 1.4 8l4-4v12.7h2V4l4.3 4L13 6.4z"></path>
+          </svg>Back to top
+        </a>
+      </div>
+    </div>
+    <!-- <div class="govuk-grid-column-one-third" aria-label="sidebar">
```
Comment (jon-shipley) [inline admin/views/helpdesk/school-summary.ejs:124]: And this?

File: admin/services/data-access/school-summary.data.service.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    SELECT
+      COUNT(p.id) as [TotalCount],
+      SUM(CASE WHEN p.checkComplete = 1 AND p.attendanceId IS NULL THEN 1 ELSE 0 END) as [Completed],
+      SUM(CASE WHEN p.attendanceId IS NOT NULL THEN 1 ELSE 0 END) as [NotAttending],
+      MIN(s.name) as [schoolName],
+      MIN(s.dfeNumber) as [dfeNumber]
+    FROM
+      [mtc_admin].[pupil] p
+      INNER JOIN [mtc_admin].school s ON p.school_id = s.id
+    WHERE p.school_id = @schoolId`
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply admin/services/data-access/school-summary.data.service.js]: updated

File: admin/services/data-access/school-summary.data.service.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+const { TYPES } = require('./sql.service')
+const sqlService = require('./sql.service')
+const R = require('ramda')
+
+const service = {}
+
+/**
+ * @description fetches register summary data from SQL data store
+ * @param {number} schoolId
+ * @return {Promise<object>}
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply admin/services/data-access/school-summary.data.service.js]: typedefs created for all 3 and docs updated

File: admin/services/data-access/school-summary.data.service.js (hunk 1 above)
Comment (GuyHarwood) [reply admin/services/data-access/school-summary.data.service.js]: thanks for the suggestion 👌

**Comment by GuyHarwood** [comment]
@jon-shipley commented markup removed and other items addressed. thanks 👍

File: admin/services/school-summary.service.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+const dataService = require('./data-access/school-summary.data.service')
+
+const service = {}
+
+/**
+ * @description get check activity summary
+ * @param {number} schoolId
+ * @return {Promise<object>}
+ */
+service.getSummary = async (schoolId) => {
```
Comment (jon-shipley) [inline admin/services/school-summary.service.js:12]: I think we shouldn't use anonymous functions as it would make any use of APM later on unreadable.

Comment (GuyHarwood) [reply admin/services/school-summary.service.js]: done for service and data service

File: admin/services/data-access/school-summary.data.service.js [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+    WHERE p.school_id = @schoolId
+    GROUP BY p.school_id`
+  const result = await sqlService.readonlyQuery(sql, [schoolIdParam])
+  return R.head(result)
+}
+
+/**
+ * Pupil Register Data Object
+ * @typedef {Object} LiveCheckData
+ * @property {Date} Date - short string formatted date
```
Comment (jon-shipley) [inline admin/services/data-access/school-summary.data.service.js:47]: Again with the capital letters.

File: admin/services/data-access/school-summary.data.service.js [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+ * Pupil Register Data Object
+ * @typedef {Object} TryItOutCheckData
+ * @property {Date} Date - short string formatted date
+ * @property {number} PinsGenerated - number of pupil pins generated
+ */
+
+/**
+ * @description fetches register summary data from SQL data store
+ * @param {number} schoolId
+ * @return {Promise<TryItOutCheckData>}
```
Comment (jon-shipley) [inline admin/services/data-access/school-summary.data.service.js:89]: Pretty sure this actually returns an array of TryItOutCheckData objects

File: admin/services/data-access/school-summary.data.service.js [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+ * @property {Date} Date - short string formatted date
+ * @property {number} PinsGenerated - number of pupil pins generated
+ * @property {number} LoggedIn - number of pupils that have logged in
+ * @property {number} Complete - number of pupils that have completed the check
+ */
+
+/**
+ * @description fetches register summary data from SQL data store
+ * @param {number} schoolId
+ * @return {Promise<LiveCheckData>}
```
Comment (jon-shipley) [inline admin/services/data-access/school-summary.data.service.js:56]: Pretty sure this actually returns an array of LiveCheckData objects.

**Comment by GuyHarwood** [comment]
@jon-shipley thanks for the feedback, all items addressed

**Pull request merged**